geth/account/accounts_test.go

```diff
@@ -103,9 +107,381 @@ func TestVerifyAccountPasswordWithAccountBeforeEIP55(t *testing.T) {
+		{
+			"fail_keyStore",
+			[]interface{}{nil, testErrKeyStore},
+			[]interface{}{s.shh, nil},
+			s.address,
+			s.password,
+			true,
+		},
+		{
+			"fail_whisperService",
+			[]interface{}{s.keyStore, nil},
+			[]interface{}{nil, testErrWhisper},
+			s.address,
+			s.password,
+			true,
+		},
+		{
+			"fail_wrongAddress",
+			[]interface{}{s.keyStore, nil},
+			[]interface{}{s.shh, nil},
+			"wrong-address",
+			s.password,
+			true,
+		},
+		{
+			"fail_wrongPassword",
+			[]interface{}{s.keyStore, nil},
+			[]interface{}{s.shh, nil},
+			s.address,
+			"wrong-password",
+			true,
+		},
+	}
+
+	for _, testCase := range testCases {
+		s.T().Run(testCase.name, func(t *testing.T) {
+			s.reinitMock()
+			s.nodeManager.EXPECT().AccountKeyStore().Return(testCase.accountKeyStoreReturn...).AnyTimes()
+			s.nodeManager.EXPECT().WhisperService().Return(testCase.whisperServiceReturn...).AnyTimes()
+			err := s.accManager.SelectAccount(testCase.address, testCase.password)
+			if testCase.fail {
+				s.Error(err)
+			} else {
+				s.NoError(err)
+			}
+		})
+	}
+}
+
+func (s *ManagerTestSuite) TestCreateChildAccount() {
+	s.reinitMock()
+
+	// First, test the negative case where an account is not selected
+	// and an address is not provided.
+	s.accManager.selectedAccount = nil
+	s.T().Run("fail_noAccount", func(t *testing.T) {
+		s.nodeManager.EXPECT().AccountKeyStore().Return(s.keyStore, nil).AnyTimes()
+		_, _, err := s.accManager.CreateChildAccount("", s.password)
+		s.Error(err)
+	})
+
+	// Now, select the test account for rest of the test cases.
+	s.reinitMock()
+	s.nodeManager.EXPECT().AccountKeyStore().Return(s.keyStore, nil).AnyTimes()
+	s.nodeManager.EXPECT().WhisperService().Return(s.shh, nil).AnyTimes()
+	err := s.accManager.SelectAccount(s.address, s.password)
+	s.NoError(err)
+
+	testCases := []struct {
+		name                  string
+		address               string
+		password              string
+		accountKeyStoreReturn []interface{}
+		fail                  bool
+	}{
+		{
+			"success",
+			s.address,
+			s.password,
+			[]interface{}{s.keyStore, nil},
+			false,
+		},
+		{
+			"fail_keyStore",
+			s.address,
+			s.password,
+			[]interface{}{nil, testErrKeyStore},
+			true,
+		},
+		{
+			"fail_wrongAddress",
+			"wrong-address",
+			s.password,
+			[]interface{}{s.keyStore, nil},
+			true,
+		},
+		{
+			"fail_wrongPassword",
+			s.address,
+			"wrong-password",
+			[]interface{}{s.keyStore, nil},
+			true,
+		},
+	}
+
+	for _, testCase := range testCases {
+		s.T().Run(testCase.name, func(t *testing.T) {
+			s.reinitMock()
+			s.nodeManager.EXPECT().AccountKeyStore().Return(testCase.accountKeyStoreReturn...).AnyTimes()
+			childAddr, childPubKey, err := s.accManager.CreateChildAccount(testCase.address, testCase.password)
+			if testCase.fail {
+				s.Error(err)
+			} else {
+				s.NoError(err)
+				s.NotNil(childAddr)
+				s.NotNil(childPubKey)
+			}
+		})
+	}
+}
+
+func (s *ManagerTestSuite) TestSelectedAndReSelectAccount() {
```

Comment:
I would split `SelectAccount` and `ReselectAccount` into two tests. It's a little bit hard here to figure our all scenarios being tested. Maybe you can add comments to some of them? E.g. `// test reselect when there is no account selected` etc.

Reply:
I'll make it more clear.